enterprise/internal/batches/scheduler/window/config.go

```diff
@@ -0,0 +1,58 @@
+package window
+
+import (
+	"github.com/hashicorp/go-multierror"
+	"github.com/inconshreveable/log15"
+	"github.com/pkg/errors"
+
+	"github.com/sourcegraph/sourcegraph/internal/conf"
+	"github.com/sourcegraph/sourcegraph/schema"
+)
+
+// Configuration represents the rollout windows configured on the site.
+type Configuration struct {
+	windows []Window
+}
+
+// NewConfiguration constructs a Configuration based on the current site
+// configuration, including watching the configuration for updates.
+func NewConfiguration() *Configuration {
+	cfg := &Configuration{}
+
+	conf.Watch(func() {
+		if err := cfg.updateFromConfig(conf.Get().BatchChangesRolloutWindows); err != nil {
+			log15.Warn("ignoring erroneous batchChanges.rolloutWindows configuration", "err", err)
+		}
+	})
+
+	return cfg
+}
+
+// ValidateConfiguration validates the given site configuration.
+func ValidateConfiguration(raw *[]*schema.BatchChangeRolloutWindow) error {
+	return (&Configuration{
+		windows: []Window{},
+	}).updateFromConfig(raw)
+}
+
+func (cfg *Configuration) updateFromConfig(raw *[]*schema.BatchChangeRolloutWindow) error {
```

Comment:
Nitpick, but `cfg.updateFromConfig` is a bit weird 😄 

```suggestion
func (cfg *Configuration) updateFromRaw(raw *[]*schema.BatchChangeRolloutWindow) error {
```

Reply:
Yeah, you can really see the seams there where I did a last minute rename of a type. I'll change it to just `update`. :smile: